### ISSUE TITLE
Invalid ES6 produced when emitting function that returns js.Dynamic.literal

### ISSUE THREAD

**Issue by shadaj** [scala-js/scala-js#3926]
Code:
```scala
package tutorial.webapp

import scala.scalajs.js

object TutorialApp {
  def main(args: Array[String]): Unit = {
    println(() => js.Dynamic.literal(x = 123))
  }
}
```

To reproduce, run `sbt run` with this code in the tutorial app, fails with `SyntaxError: Unexpected token ':'`.

Scala.js version: 1.0.0-RC2
Scala version: 2.12.10 (interestingly, the bug **does not appear when using 2.13**)

Relevant emitted JS:
```js
const x = new $c_sjsr_AnonFunction0(((this$1) => (() => {
  "x": 123
}))(this));
```

Which fails to parse because the `{` is interpreted as opening the function body. One way that this could be fixed would be to emit parentheses around the value so it would look like

```js
const x = new $c_sjsr_AnonFunction0(((this$1) => (() => ({
  "x": 123
})))(this));

**Comment by sjrd** [comment]
Thanks for the report!

**Comment by sjrd** [comment]
Fixed in e00742189b249e6e424ef753a982136eb6bce95a.
